external-crates/move/crates/move-analyzer/src/analysis/typing_analysis.rs

```diff
@@ -713,32 +981,53 @@ impl<'a> TypingVisitorContext for TypingAnalysisContext<'a> {
                     true
                 }
                 TE::Pack(mident, name, tyargs, fields) => {
-                    // add use of the struct name
-                    visitor.add_datatype_use_def(mident, name, &name.loc());
+                    visitor.add_datatype_use_def(mident, name);
                     for (fpos, fname, (_, (_, init_exp))) in fields.iter_mut() {
-                        // add use of the field name
-                        visitor.add_field_use_def(&mident.value, &name.value(), fname, &fpos);
-                        // add field initialization expression
+                        visitor.add_struct_field_use_def(
+                            &mident.value,
+                            &name.value(),
+                            fname,
+                            &fpos,
+                        );
                         visitor.visit_exp(init_exp);
                     }
-                    // add type params
-                    for t in tyargs.iter_mut() {
-                        visitor.visit_type(Some(exp_loc), t);
-                    }
+                    tyargs
+                        .iter_mut()
+                        .for_each(|t| visitor.visit_type(Some(exp_loc), t));
                     true
                 }
                 TE::Borrow(_, exp, field) => {
                     visitor.visit_exp(exp);
-                    visitor.add_field_type_use_def(&exp.ty, &field.value(), &field.loc());
+                    visitor.add_struct_field_type_use_def(&exp.ty, &field.value(), &field.loc());
+                    true
+                }
+                TE::PackVariant(mident, name, vname, tyargs, fields) => {
```

Comment:
This _should_ be unreachable.

Reply:
If we `debug_assert` it to `false` (much like we do for `TE::VariantMatch` right below), the current tests will fail on this assertion